### PR TITLE
Make migrations and compression methods synchronised

### PR DESCRIPTION
**Pull request by emanuelez**
Changelog:
 * Made migrations and compression methods synchronised

### CODE REVIEW

**Comment by emanuelez** [comment]
@cmelchior @kneth @bmunkholm

**Comment by kneth** [comment]
Sounds like a good idea to me.

:+1:

File: realm/src/main/java/io/realm/Realm.java [hunk 1]
```diff
@@ -1413,12 +1412,12 @@ public static void migrateRealmAtPath(String realmPath, byte[] key, RealmMigrati
...
     }
 
-    @SuppressWarnings("UnusedDeclaration")
     public static void migrateRealmAtPath(String realmPath, RealmMigration migration, boolean autoRefresh) {
         migrateRealmAtPath(realmPath, null, migration, autoRefresh);
     }
 
-    public static void migrateRealmAtPath(String realmPath, byte[] key, RealmMigration migration, boolean autoUpdate) {
+    public static synchronized void migrateRealmAtPath(String realmPath, byte[] key, RealmMigration migration,
+                                            boolean autoUpdate) {
```
Comment (bmunkholm) [inline realm/src/main/java/io/realm/Realm.java]: any system in that indentation?

**Comment by bmunkholm** [comment]
:+1:

File: realm/src/main/java/io/realm/Realm.java (hunk 1 above)
Comment (emanuelez) [inline realm/src/main/java/io/realm/Realm.java]: Yes. Max 120 chars lines. That's the convention I thought we were using. I set up Studio to split on that.

**Pull request merged**